Week3/homework/7-step3.js

```diff
@@ -21,5 +21,10 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+// In this article is explained easily. 
+//https://codeburst.io/javascript-passing-by-value-vs-reference-explained-in-plain-english-8d00fd06a47c
+
+//Basically it says that when we have two primitives, copies are made by the system. If we change the value of-
+//one of the original primitives, those will affect the outcome of only those, and not that of the copies, since
+//the copies are not aware of the changes. Hence two different results will come out.
```

Comment:
Okay that is pass by value.  
What sort of values are passed by value? i.e. which ones are copied?
What sort of values are passed by reference?
```
let myobj = {
  x: 'abc',
  y: 1
};

// define a function that uses myobj
function double(obj) {
  obj.x = obj.x + obj.x;
  obj.y = obj.y + obj.y;
  return obj;
}
// call the double function with myobj
double(myobj);
// What are the current values of myobj.x, and myobj.y?

// and now with non compound data, i.e. strings and numbers
let str = 'something';
let num = 2;

// define a second double to work with non-objects
function doubleVal(val){
  return val + val;
}
// call the doubleVal function with our str and num variables.
double(str);
double(num);
// What will the current value of str and num be?
```
If you can answer these questions without using `console.log` you should be good to go.  If you can't experiment with console.log on the results of these function calls and the original values and see if it makes sense.